### ISSUE TITLE
Passing integer key to IntEnum dictionary causes Errors

### ISSUE THREAD

**Issue by roshanjrajan-zip** [microsoft/pyright#5362]
**Describe the bug**
For an IntEnum, if you have a dictionary mapping IntEnum members to something else, using an integer to retrieve from the dictionary gives an error.

```sh
test.py
  test.py:14:15 - error: Argument of type "Literal[1]" cannot be assigned to parameter "__key" of type "TestEnum" in function "__getitem__"
    "Literal[1]" is incompatible with "TestEnum" (reportGeneralTypeIssues)
  test.py:15:38 - error: Argument of type "Literal[2]" cannot be assigned to parameter "__key" of type "TestEnum" in function "get"
    "Literal[2]" is incompatible with "TestEnum" (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 informations
```

**Expected behavior**
I would expect the pyright to understand that an int or IntEnum works as a key.

**Code or Screenshots**

```python
from enum import IntEnum

class TestEnum(IntEnum):
    OPTION1 = 1
    OPTION2 = 2
    OPTION3 = 3

map_valid_options = {
    TestEnum.OPTION1: 100,
    TestEnum.OPTION3: 200,
}

assert 1 in map_valid_options
assert 100 == map_valid_options[1] # Error is here
assert None == map_valid_options.get(2) # Error is here
assert 200 == map_valid_options[TestEnum.OPTION3]

assert 1 == TestEnum.OPTION1
assert 2 == TestEnum.OPTION2
assert 3 == TestEnum.OPTION3
```

**VS Code extension or command-line**
Using pyright 1.1.315 from CLI

**Comment by erictraut** [comment]
Pyright is doing the right thing here from a type validation standpoint. Not surprisingly, mypy gives the same result.

The key type in a `dict` is invariant, so `dict[TestEnum, int]` is not type compatible with `dict[int, int]` even though `TestEnum` derives from `int`.

If you want to treat `map_valid_options` as a `dict[int, int]`, you can declare it as such when it is first assigned.

```python
map_valid_options: dict[int, int] = {
    TestEnum.OPTION1: 100,
    TestEnum.OPTION3: 200,
}
```
